File: fibonachi.R
fib <- function(x) {
  return(if(x < 2) x else fib(x - 1) + fib(x - 2))
}
max_number <- 38
if(length(commandArgs()) > 5) max_number <- as.integer(commandArgs()[6])

for(i in 1:max_number) {
  print(paste(i, "-", fib(i)))
}

File: fibonachi.lua
function fib(x)
  if x < 2 then return(x) else return(fib(x - 1) + fib(x - 2)) end
end

max_number = 38
if arg[1] then max_number = tonumber(arg[1]) end
for i = 0, max_number do
  print(i .. ' - ' .. fib(i))
end
